OIDC and OIDC Client: Support JWT bearer client authentication using client assertion loaded from filesystem (#45131)
quarkusio/quarkus: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc

Using `client_secret_jwt` or `private_key_jwt` authentication methods ensures that a client secret does not get sent to the OIDC provider, therefore avoiding the risk of a secret being intercepted by a  'man-in-the-middle' attack.

.Example how JWT Bearer token can be used to authenticate client

[source,properties]
----
quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
quarkus.oidc.client-id=quarkus-app
quarkus.oidc.credentials.jwt.source=bearer <1>

> quarkus.oidc.credentials.jwt.jwt-bearer-token-path=/var/run/secrets/tokens <2>

Reviewer: ```suggestion
quarkus.oidc.credentials.jwt.token-path=/var/run/secrets/tokens <2>
```

Sorry Michal, I  know we briefly discussed the property name and a `jwt-bearer-token-path` is totally fine, but once I looked at the complete config example, I thought there was a bit of duplication, both for `jwt` and `bearer` :-)

Author: I agree, the `token-path` sounds better. Thanks.

----

> <1> The link:https://www.rfc-editor.org/rfc/rfc7523[RFC7523] explains how JWT Bearer tokens can be used to authenticate clients, see the link:https://www.rfc-editor.org/rfc/rfc7523#section-2.2[Using JWTs for Client Authentication] section for more information.

Reviewer: ```suggestion
<1> Use JWT Bearer token to authenticate clients, see the link:https://www.rfc-editor.org/rfc/rfc7523#section-2.2[Using JWTs for Client Authentication] section for more information.
```

> <2> Path to a JWT Bearer token. Quarkus loads a new token from a filesystem when necessary according to the token expiration.

Reviewer: ```suggestion
<2> Path to a JWT bearer token. Quarkus loads a new token from a filesystem and reloads it when the token has expired.
```
